docs/validation.md

```diff
@@ -133,29 +133,6 @@ simply set:
 yahooFinance.setGlobalConfig({ validation: { logErrors: false} });
 ```
 
-<a name="note-additional-props"></a>
-## A Note on "Additional Properties"
-
-*Since v2.2 (March 2022)*
-
-We now allow "additional properties" on results received from Yahoo.
-
-This means that while we can still guarantee the type of all known fields we
-receive back, we'll no longer throw a new error on new unknown keys.
-
-This is important because Yahoo constantly add new fields, and this would
-break all existing deployments.
-
-You can revert to the old behaviour with:
-
-```js
-yahooFinance._disallowAdditionalProps();
```

Comment:
This is something I would actually like to keep, in one form or another.  Doesn't necessarily need identical API since we prefixed with `_`.

* For the **user**, most will never need this unless they're pedantic.  We don't need to cover this.

* For **contributors**, this has been the most useful way to see if we're adequately covering the entire response (that's why it's the default for `NODE_ENV=test`.  I don't really want to accept commits unless they actually provide full typing for everything Yahoo is supplying - otherwise tests should fail.

So, the big requirement here is more for test validation during development rather than regular runtime validation of the library.

Reply:
Ok not an issue.

I've added this as an option to the `options.validation` field and set it to default to `NODE_ENV === "test"` so that covers off the need for stricter types during development, whilst giving the options for callers to specify it themselves where it's useful to them with the caveat that it's an internal only option.

Let me know if this is sufficient to support all parties, if not I can have a bit more of a think about it

Reply:
Brilliant, yes, that's absolutely perfect! :pray: